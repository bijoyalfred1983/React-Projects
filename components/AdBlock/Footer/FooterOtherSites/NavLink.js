/* @flow */

import React from 'react';
import type { Element } from 'react';
import styles from './styles.scss';
import { jsTrack } from '../../../containers/Analytics/ClickTrack';

type Props = { NavLinks: Array<Object> };

export const trackFooterLink = (e, linkName) => {
  jsTrack.trackFooterNavMenu(linkName);
};

const NavLink = ({ NavLinks }: Props): Element<'div'> => (
  <div className={styles.footerOther}>
    <div className={styles.listHolder}>
      <ul className={styles.list}>
        {NavLinks.map(navlink => (
          <li key={navlink.id}>
            <a
              href={navlink.URL}
              onClick={e => trackFooterLink(e, navlink.content)}
            >
              {navlink.content}
            </a>
          </li>
        ))}
      </ul>
    </div>
  </div>
);

export default NavLink;
