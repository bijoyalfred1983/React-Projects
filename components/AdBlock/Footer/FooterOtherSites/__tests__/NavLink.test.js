/* @flow */

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavLink from '../NavLink';

describe('<NavLink />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      NavLinks: [
        {
          id: '1',
          content: 'AutoZoner Services '
        },
        {
          id: '2',
          content: 'AutoZoner Benefits Login'
        },
        {
          id: '3',
          content: 'AutoZone Pro'
        },
        {
          id: '4',
          content: 'ALLDATAdiy'
        },
        {
          id: '5',
          content: 'ALLDATA Repair'
        }
      ]
    };
  });

  test('renders on footerOtherSites NavLink correctly', () => {
    const wrapper = mount(
        <NavLink {...PROPS} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


