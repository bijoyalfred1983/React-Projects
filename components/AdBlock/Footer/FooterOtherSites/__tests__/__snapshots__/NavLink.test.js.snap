// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLink /> renders on footerOtherSites NavLink correctly 1`] = `
<NavLink
  NavLinks={
    Array [
      Object {
        "content": "AutoZoner Services ",
        "id": "1",
      },
      Object {
        "content": "AutoZoner Benefits Login",
        "id": "2",
      },
      Object {
        "content": "AutoZone Pro",
        "id": "3",
      },
      Object {
        "content": "ALLDATAdiy",
        "id": "4",
      },
      Object {
        "content": "ALLDATA Repair",
        "id": "5",
      },
    ]
  }
>
  <div
    className="footerOther"
  >
    <div
      className="listHolder"
    >
      <ul
        className="list"
      >
        <li>
          <a
            href="#!"
          >
            AutoZoner Services 
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
            AutoZoner Benefits Login
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
            AutoZone Pro
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
            ALLDATAdiy
          </a>
        </li>
        <li>
          <a
            href="#!"
          >
            ALLDATA Repair
          </a>
        </li>
      </ul>
    </div>
  </div>
</NavLink>
`;
